draft-kazuho-httpbis-priority.md

```diff
@@ -302,7 +307,71 @@ extension element (see {{!RFC7540}}, Section 5.5).
 
 # Considerations
 
-## Why use an End-to-End Header Field?
+## The Case for a Collaborative Prioritization Process {#collaboration}
+
+HTTP/2 prioritization places a lot of responsibility on the client. It provides
+a very flexible dependency-priority scheme that has the potential to model rich
+and diverse use cases. However, experience shows that HTTP/2 implementations
+have not used this for much innovation, instead preferring simpler structures.
+
+The client is well placed to decide request priority based on information such
+as the resource type and the context it will be used in. The goal and how
+information is acted on is described lightly in {{?RFC7540}} Section 5.3:
+
+>  Most importantly, priority can be used to select streams for transmitting
+   frames when there is limited capacity for sending.
+
+> Explicitly setting the priority for a stream is input to a
+  prioritization process.  It does not guarantee any particular
+  processing or transmission order for the stream relative to any other
+  stream.  An endpoint cannot force a peer to process concurrent
+  streams in a particular order using priority.  Expressing priority is
+  therefore only a suggestion.
+
+The client is not alone in understanding the best way to prioritize a request. A
+server's prioritization decision is driven by a number of factors such as the
+selected representation ({{?RFC7231}}, Section 3), data locality, resource
+contention, load balancing across multiple clients, application-specific
+annotations etc. When it generates responses it might determine that the
+client's predicted prioritization was incorrect for how the selected
+representation will be used. Rigidly following the client's suggestion could
```

Comment:
I appreciate the point you are making in this and the next comment. I think the general problem with the text as it stands is that it tries to describe how HTTP/2 works today, *and* make the case for an explicit server-driven priority signal.

To the first point I was attempting to describe a case we have today in HTTP/2 where a normal server gets a a request that is obviously a dumb prioritisation e.g. let's pick on Spartan-Edge. By ignoring the suggestion, we can improve page load time.

However, for the above case, no explicit signal is required and I think therefore that the argument is not strong (as you point out "sending the priority response header to the client does not affect how the responses are prioritized."). 

> Such an intermediary just obeys to the origin's instruction, rather than determining something as wrong.

I think this is the root to one of the problems with this PR. Is it your opinion that collaboration of initial priority involves only  the client and the origin? And that the intermediary cannot override parameters, all it can do is use the priority information in combination with other factors in order to manage the resource prioritization on the HTTP/2 hop to the client.

Reply:
I think these were points that @rmarx was trying to make in a previous review too.